pkg/clustermesh/selectbackends.go

```diff
@@ -51,19 +52,21 @@ func (sb ClusterMeshSelectBackends) SelectBackends(bes iter.Seq2[experimental.Ba
 				continue
 			}
 			if be.Source == source.ClusterMesh {
-				localBackends++
-			} else {
 				remoteBackends++
+			} else {
+				localBackends++
 			}
 		}
 		switch affinity {
 		case experimental.ServiceAffinityLocal:
-			useRemote = localBackends == 0
+			// Always include the local backends even if they are unhealthy and
+			// only include (healthy) remote ones if there are no healthy local backends.
+			useLocal = true
+			useRemote = localBackends == 0 && remoteBackends > 0
```

Comment:
Should we also include the remote backends if all the local ones are terminating? Otherwise I guess new connections would fail during that period of time.

Reply:
Ah good point. Changed.